.github/workflows/ci.yml

```diff
@@ -478,3 +478,22 @@ jobs:
           files: py-upgradability.json
           fail_ci_if_error: true
           flags: pytests,upgradability,linux
+  
+  public_libraries_compilation:
+    name: "${{matrix.name}} Public libraries compilation"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Linux
+            os: ubuntu-22.04-16core
+          - name: Macos
+            os: macos-latest-xlarge
+          - name: Windows
+            os: windows-latest-xlarge
+    timeout-minutes: 90
+    steps:
+      - uses: actions/checkout@v4
+      # Run the build for public facing libraries
+      - run: cargo build --release -p near-primitives -p near-crypto -p near-jsonrpc-primitives -p near-chain-configs -p near-primitives-core
```

Comment:
I would probably replace this with a `cargo check` (I don't think `cargo build --release` checks anything particularly much extra as far as compiling for Windows is concerned), and while at that we can define what's a public library in the `Justfile`.

This then also doesn't need machines as large as specified & Linux/MacOS are already not only `check`ed but also fully tested in another job(s).

Reply:
Done